src/radical/pilot/agent/scheduler/base.py

```diff
@@ -272,6 +273,10 @@ def initialize(self):
         self._p.daemon = True
         self._p.start()
 
+        self._pwatcher = ru.PWatcher(uid='%s.pw' % self._uid, log=self._log)
+        self._pwatcher.watch(os.getpid())
+        self._pwatcher.watch(self._p.pid)
+
```

Comment:
quick check - do we need this part considering that the base component already tracks itself?

Reply:
It's not about watching itself (that's not really possible - if `self` goes away, so does the watcher. Instead, this is tying the self process to the lifetime of `self._p` - if the latter dies, self will also be killed.

Reply:
ok, got it! thanks!